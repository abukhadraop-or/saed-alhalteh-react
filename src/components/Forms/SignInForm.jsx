import React, { useState } from 'react';
import Input from 'components/Forms/Input';
import { FormContainer, Title, RedirectLink, SubmitButton } from './sign-in-form-styles';

function SignInForm() {
  const [userData, setUserData] = useState({ email: '', password: '' });
  function handleSubmit(submitEvent) {
    submitEvent.preventDefault();
  }

  function handleChange(changeEvent) {
    const userDataClone = { ...userData };
    userDataClone[changeEvent.target.name] = changeEvent.target.value;
    setUserData(userDataClone);
  }
  return (
    <FormContainer>
      <form onSubmit={(submitEvent) => handleSubmit(submitEvent)}>
        <Title> Sign In </Title>
        <RedirectLink to="/signUp"> Need an Account? </RedirectLink>
        <Input
          value={userData.email}
          onChange={(changeEvent) => handleChange(changeEvent)}
          name="email"
          type="text"
          placeholder="Email"
        />
        <Input
          value={userData.password}
          onChange={(changeEvent) => handleChange(changeEvent)}
          name="password"
          type="password"
          placeholder="Password"
        />
        <SubmitButton type="submit"> Sign In </SubmitButton>
      </form>
    </FormContainer>
  );
}
export default SignInForm;
