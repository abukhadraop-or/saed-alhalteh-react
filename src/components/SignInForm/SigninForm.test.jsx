import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import SignInForm from '.';

describe('<SignInForm /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<SignInForm />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
