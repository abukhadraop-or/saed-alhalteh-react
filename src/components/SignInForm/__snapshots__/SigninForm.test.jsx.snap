// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> tests matches the snapshot 1`] = `
<h1>
  SignInForm
</h1>
`;
